### PR TITLE
Clickable advice for setting max number of failures

### LINKED ISSUE #1628: Advice to use `Sys.setenv('TESTTHAT_MAX_FAILS' = Inf)` should be clickable
i.e. here https://github.com/r-lib/testthat/blob/0f24eae7bd2cd7d0fbe5a5492636731d66d0dd26/R/reporter-progress.R#L211

### PR DESCRIPTION
**Pull request by hadley**
Fixes #1628

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -155,6 +155,7 @@ export(not)
 export(prints_text)
 export(quasi_label)
 export(run_cpp_tests)
+export(set_max_fails)
```
Comment (hadley) [inline NAMESPACE:158]: Not sure about this name

File: R/reporter-progress.R [hunk 2]
```diff
@@ -540,3 +540,16 @@ skip_bullets <- function(skips) {
   tbl <- table(skips)
   paste0(cli::symbol$bullet, " ", names(tbl), " (", tbl, ")")
 }
+
+
+#' Set maximum number of test failures before aborting the run
```
Comment (DavisVaughan) [inline R/reporter-progress.R:545]:
```suggestion
#' Set maximum number of test failures allowed before aborting the run
```

**Review by DavisVaughan — APPROVED** [review]
Name sounds fine to me, it is in the same vein as `testthat::set_reporter()`.

I don't _think_ we need `get_max_fails()`, but wanted to mention it in case you think we do

File: R/reporter-zzz.R [hunk 3]
```diff
@@ -59,7 +59,7 @@ with_reporter <- function(reporter, code, start_end_reporter = TRUE) {
 
 stop_reporter <- function(message) {
   signal(message, "testthat_abort_reporter")
```
Comment (DavisVaughan) [inline R/reporter-zzz.R:61]: Is signaling a `message` with cli inline markup problematic at all?

Comment (hadley) [reply R/reporter-zzz.R:61]: I don't think anything actually uses the message, so might be safer to set to `""`.

Comment (DavisVaughan) [reply R/reporter-zzz.R:61]: @lionel- could `signal()` allow `message` to be missing, i.e. `signal(class = "testthat_abort_reporter")`? I also just had a case in dplyr where I just wanted to do the signaling but didn't care about the message

Comment (lionel-) [reply R/reporter-zzz.R:61]: I think it'd be fine to take `""` by default.

Comment (hadley) [reply R/reporter-zzz.R:61]: Actually, I can just combine the `cli_abort()` and `signal()` — I think this must be a remnant from when the code used `stop()` and it was harder to class the error.

**Pull request merged**